.github/workflows/npe-x500-m3.yml

```diff
@@ -0,0 +1,43 @@
+name: NPE X500 M3
+
+on:
+  # With these triggers the Yocto jobs will run
+  # in parallel with the Flowzone jobs, which is fine for now
+  # and allows us to better control what we want to test and when.
+  # It is expected that Flowzone could fail, but yocto jobs will run.
+  pull_request:
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    branches:
+      - "main"
+      - "master"
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+\+?r?e?v?*
+      - v20[0-9][0-9].[0-1]?[1470].[0-9]+
```

Comment:
I think we need to talk about these two regex, and how to ensure that only one of them matches an ESR tag and is used for ESR devices (here and in the workflow itself). Before we copy this pattern everywhere and it becomes more work to change.

Reply:
This was copied from the deploy scripts originally: https://github.com/balena-os/github-workflows/blob/master/.github/workflows/build_and_deploy.yml#L67 for context for all reading. 

There is also the existing workflow on device repos: https://github.com/balena-os/balena-raspberrypi/blob/master/.github/workflows/build_and_deploy.yml#L4C2-L7C42 that use it.

Looking at the regexp again, with https://regex101.com/ open in another tab to check:

1. `v[0-9]+.[0-9]+.[0-9]+\+?r?e?v?*` - shouldn't this have `v[0-9]+.[0-9]+.[0-9]+\+?r?e?v?*[0-9]+?` ? The regexp checker doesn't seem to like the `*` wildcard at the end, but its worked so far in gha ?
2. an esr version tag, e.g `v2020.0.1` would still trigger the `v[0-9]+.[0-9]+.[0-9]+\+?r?e?v?*` regexp wouldn't it?
3.  If so, that will stop our method of commenting out the esr regexp in the device workflow file to stop esr deploys
4. Looking at the workflow again, https://github.com/balena-os/balena-yocto-scripts/blob/master/.github/workflows/yocto-build-deploy.yml#L109 this is not a sage way to determine `deploy-esr` is it? As a balenaOS of version `v20.Y.Z+rev?` will make that evaluate to true

Reply:
I'm actually thinking something like this will be required
```yml
  push:
    # https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore
    # https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
    branches:
      - main
      - master
      # ESR branches glob pattern
      - v20[0-9][0-9].[0-1]?[1470].[0-9]+
    tags:
      # Semver tags glob pattern (includes ESR in format v20YY.MM.PATCH)
      - v[0-9]+.[0-9]+.[0-9]+\+?r?e?v?*
```
Then in the workflow we check the branch name to enable/disable ESR, but still expose it as an input for overrides.

Reply:
ah that feels a lot cleaner @klutchell  - and for device types we don't want to deploy esr's for, we comment out/remove the branch?

Using the actual branch name definitely feels less flimsy than using the tag

Reply:
Correct, we remove the 3rd branch trigger for non-esr devices. We just need to update the workflow to check the branch name not the tag because ESR tags are semver compliant already on purpose.

Reply:
I pushed these suggestions to this PR: https://github.com/balena-os/balena-intel/pull/756

Reply:
@klutchell @rcooke-warwick Maybe missing context but, 
If we are deciding to check branch names and not tags (which I agree is less flimsy) then I recommend we change the flowzone.yml as well https://github.com/balena-os/balena-intel/blob/master/.github/workflows/flowzone.yml

Which is using a similar glob pattern albeit missing a few more specifications `'20[0-9][0-9].[0-1]?[1470].x'`

Reply:
We revisited this on the call and the logic is much more complex unfortunately.

I will post an updated proposal when I have something to review.